### PR TITLE
Enable max length for textarea field

### PR DESCRIPTION
**Pull request by phalkunz**
Issue: https://github.com/silverstripe/silverstripe-framework/issues/7549

### CODE REVIEW

File: src/Forms/TextField.php [hunk 1]
```diff
@@ -79,6 +79,17 @@ public function getAttributes()
         );
     }
 
+    public function getSchemaDataDefaults() {
```
Comment (robbieaverill) [inline src/Forms/TextField.php]: You'll need to drop the opening brace to a new line - see the linting failures on https://travis-ci.org/silverstripe/silverstripe-framework/jobs/311519327

**Comment by robbieaverill** [comment]
I've fixed the Scrutinizer configuration here: https://github.com/silverstripe/silverstripe-framework/pull/7678

File: src/Forms/TextField.php (hunk 1 above)
Comment (phalkunz) [reply src/Forms/TextField.php]: Thanks @robbieaverill. Will fix that.

File: src/Forms/TextareaField.php [hunk 2]
```diff
@@ -45,19 +45,29 @@ class TextareaField extends FormField
...
         $data = parent::getSchemaDataDefaults();
-        
+
         $data['data']['rows'] = $this->getRows();
         $data['data']['columns'] = $this->getColumns();
-        
+
+        $maxLength = $this->getMaxLength();
+        if ($maxLength) {
+            $data['maxLength'] = $maxLength;
```
Comment (tractorcow) [inline src/Forms/TextareaField.php]: Top level form schema keys should only be for properties that apply to all fields; maxLength only makes sense to text-based fields.

This should go into `$data['attributes']['maxLength']` or `$data['data']['maxLength']`

File: src/Forms/TextareaField.php [hunk 3]
```diff
@@ -106,6 +116,25 @@ public function getColumns()
...
+    {
+        $this->maxLength = $maxLength;
+
+        return $this;
+    }
+
+    /**
+     * @return null|int
+     */
+    public function getMaxLength()
```
Comment (tractorcow) [inline src/Forms/TextareaField.php]: TextField adds this to `getAttributes`. Shouldn't be do this in TextareaField also?

**Comment by phalkunz** [comment]
@tractorcow Thanks for the feedbacks :) I've updated the pull request.

**Comment by tractorcow** [comment]
@phalkunz can you please update the unit tests?

File: src/Forms/TextareaField.php [hunk 4]
```diff
@@ -117,7 +140,8 @@ public function getAttributes()
                 'rows' => $this->getRows(),
                 'cols' => $this->getColumns(),
                 'value' => null,
-                'type' => null
+                'type' => null,
+                'maxlength' => $this->getMaxLength(),
```
Comment (tractorcow) [inline src/Forms/TextareaField.php]: We should only set this attribute if non empty.

See TextField.php for pre-existing logic:

```php
if ($maxLength) {
    $attributes['maxLength'] = $maxLength;
    $attributes['size'] = min($maxLength, 30);
}
```

Comment (phalkunz) [reply src/Forms/TextareaField.php]: Fixed.

**Comment by tractorcow** [comment]
I've just pushed up a fix to linting.

**Pull request merged**